avahi-core/server.c

```diff
@@ -1626,30 +1638,42 @@ AvahiServerConfig* avahi_server_config_copy(AvahiServerConfig *ret, const AvahiS
             return NULL;
 
     if (c->domain_name)
-        if (!(d = avahi_strdup(c->domain_name))) {
-            avahi_free(h);
-            return NULL;
+        if (!(d = avahi_strdup(c->domain_name)))
+            goto clean;
+
+    if (!(browse = avahi_string_list_copy(c->browse_domains)) && c->browse_domains)
+        goto clean;
+
+    if (!(allow = avahi_string_list_copy(c->allow_interfaces)) && c->allow_interfaces)
+        goto clean;
+
+    if (!(deny = avahi_string_list_copy(c->deny_interfaces)) && c->deny_interfaces)
+        goto clean;
+
+    if (c->allow_addresses) {
```

Comment:
Similarly, please create static function ``avahi_address_list_copy`` with contents from this list, which woud get called from the body of this if for allow_addresses and deny_addresses. I think the code for them would be same, right?